src/TTImp/Elab/App.idr

```diff
@@ -330,6 +335,20 @@ mutual
                       else pure $ dotTerm tm
            else pure tm
     where
+      
+      ||| Find the constructors of a normal form that is a fully
+      ||| applied type constructor
+      possibleConstructors : NF vars -> Core (List Name)
+      possibleConstructors (NApp _ (NRef (TyCon _ n) tycName) args) = 
+        if length args == n
+        then do defs <- get Ctxt
+                Just gdef <- lookupCtxtExact tycName (gamma defs)
+                | Nothing => pure []
+                let (TCon _ _ _ _ _ _ datacons _) = gdef.definition                                                   | _ => pure []
```

Comment:
(Not related to the test failure and a bit nitpicky, but I was initially confused here by only having one case for `TCon _ ...` since the alternative was way off to the right. I usually prefer to put the alternatives on the next line, slightly indented.)

Reply:
Ooops! I'll fix.